File: t/PACSFramework/VB/StorageServerManager/UI/Authentication/LoginDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace Leadtools.Demos.StorageServer.UI.Authentication
   Partial Friend Class LoginDialog : Inherits Form
      Public Event AuthenticateUser As EventHandler(Of AuthenticateUserEventArgs)

      Public Property Username() As String
         Get
            Return textBoxUserName.Text
         End Get
         Set(ByVal value As String)
            textBoxUserName.Text = value
         End Set
      End Property

      Public Property Info() As String
         Get
            Return labelInfo.Text
         End Get
         Set(ByVal value As String)
            labelInfo.Text = value
            labelInfo.Visible = Not String.IsNullOrEmpty(value)
         End Set
      End Property

      Public Property CanSetUserName() As Boolean
         Get
            Return textBoxUserName.Enabled
         End Get
         Set(ByVal value As Boolean)
            textBoxUserName.Enabled = value
         End Set
      End Property

      Public Sub New()
         InitializeComponent()
      End Sub

      Private Sub buttonLogin_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonLogin.Click
         Dim ea As AuthenticateUserEventArgs = New AuthenticateUserEventArgs(textBoxUserName.Text, Extensions.ToSecureString(textBoxPassword.Text))

         RaiseEvent AuthenticateUser(Me, ea)
         If ea.Cancel Then
            DialogResult = DialogResult.Cancel
         End If
         If ea.InvalidCredentials Then
            DialogResult = DialogResult.None
            textBoxPassword.Text = String.Empty
            textBoxUserName.Focus()
         End If
      End Sub

      Private Sub Credentials_Changed(ByVal sender As Object, ByVal e As EventArgs) Handles textBoxUserName.TextChanged, textBoxPassword.TextChanged
         buttonLogin.Enabled = textBoxUserName.Text.Length > 0 AndAlso textBoxPassword.Text.Length > 0
      End Sub

      Private Sub LoginDialog_Shown(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Shown
         If textBoxUserName.Text.Length > 0 Then
            textBoxPassword.Focus()
         End If
      End Sub
   End Class
End Namespace

File: t/PACSFramework/CS/DicomHighlevelPatientUpdaterDemo/Authentication/LoginDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Security;

namespace DicomDemo.Authentication
{
   public partial class LoginDialog : Form
   {
      public event EventHandler<AuthenticateUserEventArgs> AuthenticateUser = delegate { };      

      public string Username 
      {
         get
         {
            return textBoxUserName.Text;
         }
      }

      public LoginDialog()
      {
         InitializeComponent();
      }

      private void buttonLogin_Click(object sender, EventArgs e)
      {
         AuthenticateUserEventArgs ea = new AuthenticateUserEventArgs(textBoxUserName.Text, textBoxPassword.Text.ToSecureString());

         AuthenticateUser(this, ea);
         if (ea.Cancel)
            DialogResult = DialogResult.Cancel;
         if (ea.InvalidCredentials)
         {
            DialogResult = DialogResult.None;
            textBoxPassword.Text = string.Empty;
            textBoxUserName.Focus();
         }
      }

      private void Credentials_Changed(object sender, EventArgs e)
      {
         buttonLogin.Enabled = textBoxUserName.Text.Length > 0 && textBoxPassword.Text.Length > 0;
      }
   }
}
